### ISSUE TITLE
`-Zunpretty=expanded` does not emit `dep-info` anymore (1.68 -> 1.69)

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust#112898]
```sh
rm -f x.d && echo 'fn main() {}' |
    RUSTC_BOOTSTRAP=1 rustup run 1.68.2 rustc - --emit=dep-info=x.d -Zunpretty=expanded >/dev/null &&
    cat x.d
x.d: 
```

```sh
rm -f x.d && echo 'fn main() {}' |
    RUSTC_BOOTSTRAP=1 rustup run 1.69.0 rustc - --emit=dep-info=x.d -Zunpretty=expanded >/dev/null &&
    cat x.d
cat: x.d: No such file or directory
```

Related: https://github.com/rust-lang/rust/issues/93201 (i.e. having `expanded` as an actual `--emit` would be nice).

We use `-Zunpretty=expanded` in the kernel to provide the expanded output in `.rsi` outputs/targets. The dependency information is not critical, but the change made our Make target fail (so we need a change on the kernel side), and it would be nice nevertheless to have the dependency information (assuming it is accurate), e.g. we also use it for similar outputs like `.ll` and `.s`.

**Comment by apiraino** [comment]
Bisection seems to lead to commit d7948c843de94245c794e8c63dd4301a78bb5ba3 but I am not sure / cant say if really related. The mentioned issue #93201 was opened before that commit was merged :thinking: @oli-obk @petrochenkov wdy?

**Comment by oli-obk** [comment]
This looks like another symptom of #93201

We'll likely keep regressing this by accident without a test. I don't know why #93201 was fixed between when it was reported and 1.68

**Comment by ojeda** [comment]
Thanks for fixing this @Urgau!e
